src/Modules/Verto/webrtc/BaseCall.ts

```diff
@@ -113,283 +194,398 @@ export default abstract class BaseCall implements IWebRTCCall {
   }
 
   hangup(params: any = {}, execute: boolean = true) {
-    this.cause = params.cause || 'NORMAL_CLEARING'
-    this.causeCode = params.causeCode || 16
-    this.setState(State.Hangup)
+    this.cause = params.cause || 'NORMAL_CLEARING';
+    this.causeCode = params.causeCode || 16;
+    this.sipCode = params.sipCode || null;
+    this.sipReason = params.sipReason || null;
+    this.sipCallId = params.sip_call_id || null;
```

Comment:
The only change that I did in this file were here 199,200 and 201. 
The rest was running eslint and prettier.